### PR TITLE
Improve CXCancellation pass readability

### PR DESCRIPTION
**Pull request by lcapelluto**
### Summary

My intern is working on a related issue, so I noticed there was some extra code in CXCancellation that has no effect.

### Details and comments

Seems it might be required after all, I'd like to find out why. Test errors do not make it clear

Figured out what was going on, added comments, tests, and renamed to improve readability and explain what was going on

### CODE REVIEW

File: test/python/transpiler/test_cx_cancellation.py [hunk 1]
```diff
@@ -45,3 +45,73 @@ def test_pass_cx_cancellation(self):
         resources_after = out_circuit.count_ops()
 
         self.assertNotIn("cx", resources_after)
+
+    def test_pass_cx_cancellation_intermixed_ops(self):
+        """Cancellation shouldn't be effected by the order of ops on different qubits."""
```
Comment (jakelishman) [inline test/python/transpiler/test_cx_cancellation.py:50]:
```suggestion
        """Cancellation shouldn't be affected by the order of ops on different qubits."""
```

File: test/python/transpiler/test_cx_cancellation.py [hunk 2]
```diff
@@ -45,3 +45,73 @@ def test_pass_cx_cancellation(self):
...
+        circuit.cx(qr[2], qr[3])
+        circuit.cx(qr[2], qr[3])
+
+        pass_manager = PassManager()
+        pass_manager.append(CXCancellation())
+        out_circuit = pass_manager.run(circuit)
+        resources_after = out_circuit.count_ops()
+
+        self.assertIn("cx", resources_after)
+        self.assertEqual(resources_after["cx"], 3)
```
Comment (jakelishman) [inline test/python/transpiler/test_cx_cancellation.py:85]: Would it be possible to test against the exact forms of the circuits for simple cases, not just the instruction counts?

**Review by jakelishman — COMMENTED** [review]
Could you change the PR title to reflect the new changes (since it ends up in the git log)?

File: test/python/transpiler/test_cx_cancellation.py (hunk 2 above)
Comment (lcapelluto) [reply test/python/transpiler/test_cx_cancellation.py:85]: Sure, probably, but where do you see that having an effect here?

Comment (lcapelluto) [reply test/python/transpiler/test_cx_cancellation.py:85]: For the ones where no change is expected, I can test that instead

Comment (lcapelluto) [reply test/python/transpiler/test_cx_cancellation.py:85]: Ah, I didn't check which test this was. I agree with you

Comment (lcapelluto) [reply test/python/transpiler/test_cx_cancellation.py]: Updated all of them! Much better now

**Review by jakelishman — APPROVED** [review]
Looks good to me - thanks for the changes!

**Pull request merged**